app/models/file_set.rb

```diff
@@ -1,4 +1,19 @@
 # Generated by hyrax:models:install
 class FileSet < ActiveFedora::Base
   include ::Hyrax::FileSetBehavior
+
+  ##
+  # Generate an encoded iiif url for this FileSet
+  # @param base_url [String] : provided by iiif_image_url_builder or iiif_info_url_builder, looks like http://localhost
+  # @return [String] : a url to the iiif server with a properly encoded identifier, looks like http://IIIFSERVER/ab%2fcd%2f12%2f24%2fz-jp2.jp2
+  def iiif_url(base_url)
```

Comment:
For S3 storage, the IIIF identifier will just be "prefix" + "s3:" + encoded s3 identifier.  For instance, `https://rais/iiif/s3:mycollection%2Ffile.jp2`

Just an FYI, not a failure - I know what you have is good for a prototype.

Reply:
For RAIS, `s3:` tells it to use s3, but then do you include the full bucket path to the file? @jechols